src/utils/fileSystemHandler.ts

```diff
@@ -9,10 +9,9 @@ import * as path from 'node:path';
 import fs from 'graceful-fs';
 import { SourcePath } from '../common/types';
 
-export function ensureFileExists(filePath: string): void {
-  fs.mkdirSync(path.dirname(filePath), { recursive: true });
-  fs.closeSync(fs.openSync(filePath, 'w'));
-}
+export const ensureFileExists = async (filePath: string): Promise<void> => {
+  await fs.promises.mkdir(path.dirname(filePath), { recursive: true });
```

Comment:
creating writeStreams will create a file.  the mkdir is still necessary to make sure it has a folder to be created in, but the open/close is pointless